dracut/30ignition/module-setup.sh

```diff
@@ -54,16 +55,19 @@ install() {
     inst_simple "$moddir/ignition-complete.target" \
         "$systemdsystemunitdir/ignition-complete.target"
 
-    mkdir -p "$initdir/$systemdsystemunitdir/ignition-complete.target.requires"
+    inst_simple "$moddir/ignition-diskful.target" \
+        "$systemdsystemunitdir/ignition-diskful.target"
 
-    # path generated by systemd-escape --path /dev/disk/by-label/root
-    install_ignition_unit coreos-gpt-setup@.service 'coreos-gpt-setup@dev-disk-by\x2dlabel-root.service'
     install_ignition_unit ignition-setup-base.service
-    install_ignition_unit ignition-setup-user.service
     install_ignition_unit ignition-disks.service
     install_ignition_unit ignition-mount.service
     install_ignition_unit ignition-files.service
-    install_ignition_unit ignition-remount-sysroot.service
+
+    # units only started when we have a boot disk
```

Comment:
Wonder if it'd be cleaner to have the units test `ConditionPathExists=!/run/ostree-live` too?  I don't have a strong opinion though.

Reply:
All of these except `ignition-remount-sysroot.service` pull in device dependencies, so they _have_ to be enabled via generator.  Conditions are only evaluated after dependencies are pulled in.